src/Analyzers/Core/CodeFixes/UseCompoundAssignment/AbstractUseCompoundAssignmentCodeFixProvider.cs

```diff
@@ -81,12 +81,12 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
                         if (diagnostic.Properties.ContainsKey(UseCompoundAssignmentUtilities.Increment))
                         {
-                            return Increment((TExpressionSyntax)leftOfAssign);
+                            return Increment((TExpressionSyntax)leftOfAssign.WithoutTrailingTrivia(), postfix: syntaxFacts.IsStatement(currentAssignment.Parent));
                         }
 
                         if (diagnostic.Properties.ContainsKey(UseCompoundAssignmentUtilities.Decrement))
                         {
-                            return Decrement((TExpressionSyntax)leftOfAssign);
+                            return Decrement((TExpressionSyntax)leftOfAssign.WithoutTrailingTrivia(), postfix: syntaxFacts.IsStatement(currentAssignment.Parent));
```

Comment:
`syntaxFacts.IsStatement` seems legit to address https://github.com/dotnet/roslyn/issues/53969#issuecomment-857908596, but maybe there is a better way to detect

> if the result of the assignment is used

There is also a lot of repetition, but I don't see a simple way to avoid that.

Reply:
What about `switch (x = x + 1)`? The assignment parent is a statement, but it should be prefix. Am I missing something?

Reply:
Yes, you are right. Maybe `IsSimpleAssignmentStatement` is better.

https://github.com/dotnet/roslyn/blob/d3e915d2dc49f4371f510add785ea011ca6c36be/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs#L1374-L1376

Reply:
I'm not sure. But things I'd make sure they are not broken:

- `switch(...)`
- `return ....`
- `yield return ....`
- `while (...)`
- `var x = ...`
- `var x = (...)`
- `bool_expr ? ... : ...`
- `MethodCall(...)`

Reply:
Instead of doing syntax analysis here, could we do that by data flow analysis?